nanorl/algorithms/dpo/data.py

```diff
@@ -0,0 +1,60 @@
+from typing import Dict, List
+
+from torch.utils.data import Dataset
+
+
+class DPODataset(Dataset):
+    def __init__(self, data: List[Dict]):
+        self.data = data
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        return self.data[idx]
+
+
+def dpo_collate_fn(batch: List[Dict], tokenizer, max_length: int = 128):
+    prompts = [item["prompt"] for item in batch]
+    chosen_responses = [item["chosen"] for item in batch]
+    rejected_responses = [item["rejected"] for item in batch]
+
+    chosen_texts = [p + c for p, c in zip(prompts, chosen_responses)]
+    rejected_texts = [p + r for p, r in zip(prompts, rejected_responses)]
+
+    chosen_tokens = tokenizer(
+        chosen_texts,
+        max_length=max_length,
+        padding=True,
+        truncation=True,
+        return_tensors="pt",
+    )
+    rejected_tokens = tokenizer(
+        rejected_texts,
+        max_length=max_length,
+        padding=True,
+        truncation=True,
+        return_tensors="pt",
+    )
+
+    prompt_tokens = tokenizer(
+        prompts, padding=False, truncation=True, add_special_tokens=True
+    )
```

Comment:
Will seem redundant, but is required to compute the length of tokens in the prompts. This is used later to determine where to ignore as the chose and rejected both consist of the prompt which is not needed in the loss computation.